test/scripts/gh-actions/setup-deps.sh

```diff
@@ -33,9 +33,60 @@ YQ_VERSION="v4.28.1"
 GATEWAY_API_VERSION="v1.2.1"
 ENVOY_GATEWAY_VERSION="v1.2.2"
 
+
 echo "Installing yq ..."
 wget https://github.com/mikefarah/yq/releases/download/${YQ_VERSION}/yq_linux_amd64 -O /usr/local/bin/yq && chmod +x /usr/local/bin/yq
 
+# [LLMInference] This is still experimental and need latest istio
+# For now, it uses helm to install istio but it will be removed once the experimental version is stable
+# ------------------------------------------------------------
+GATEWAY_API_EXPERIMENTAL_VERSION="v1.3.0"
+GATEWAY_API_EXT_VERSION="v0.3.0"
+ISTIO_HUB="gcr.io/istio-testing"
+ISTIO_HUB_VERSION="1.26-alpha.9befed2f1439d883120f8de70fd70d84ca0ebc3d"
+HELM_VERSION="v3.17.3"
+OS=$(uname | tr '[:upper:]' '[:lower:]')
+ARCH=$(uname -m)
+
+download_helm() {
+  case "$ARCH" in
+    arm64|aarch64) ARCH="arm64" ;;
+    x86_64) ARCH="amd64" ;;
+    *) echo "Unsupported architecture: $ARCH"; exit 1 ;;
+  esac
+
+  HELM_TARBALL="helm-${HELM_VERSION}-${OS}-${ARCH}.tar.gz"
+  HELM_DIR="$HOME/.local/bin"
+  HELM_PATH="$HELM_DIR/helm"
+  
+  # Create directory if it doesn't exist
+  mkdir -p "$HELM_DIR"
```

Comment:
Isn't this a little scary to do this way? I'd just make it a pre-requisite for this command